packages/hostScriptsServices/hostScripts/check_docker_network.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+network_name="dncore_network"
+max_retries=5
+retry_delay=60
+
+# Function to check and create network with retry logic
+check_and_create_network() {
+    for ((i=0; i<max_retries; i++)); do
+        if docker network inspect "$network_name" &> /dev/null; then
+            echo "Docker network '$network_name' exists."
+            docker network inspect "$network_name"
+            return 0
+        else
+            echo "ATTENTION! Docker network '$network_name' does not exist. Creating it..."
+            # Check the core docker image tag to get the version and determine the subnet to be used for creating the docker core network
+            core_version="$(docker image ls --filter reference="core.dnp.dappnode.eth" --format '{{.Tag}}' | sort -V | tail -n 1)"
+            echo "core version retrieved $core_version"
+
+            # Compare with 0.2.30 using sort
+            if printf '0.2.30\n%s\n' "$core_version" | sort -V | head -n 1 | grep -q '0.2.30'; then
+                subnet="172.33.0.0/16"
+                # core_version is greater than or equal to 0.2.30
+                echo "Core version is greater than or equal to 0.2.30, using subnet $subnet"
+            else
+                subnet="10.20.0.0/24"
+                # core_version is less than 0.2.30
+                echo "Core version is less than 0.2.30, using subnet $subnet"
+            fi
+
+            # create docker network
+            docker network create --driver bridge --subnet "$subnet" "$network_name"
```

Comment:
If this fails we might want to create the network without specifying the subnet so that we are sure it is created and then the dappmanager will be responsible for migrating it to the proper subnet

Reply:
implemented!